tower-http/src/sensitive_header.rs

```diff
@@ -0,0 +1,224 @@
+//! Middlewares that mark headers as [sensitive].
+//!
+//! [sensitive]: https://docs.rs/http/latest/http/header/struct.HeaderValue.html#method.set_sensitive
+
+use futures_util::ready;
+use http::{header::HeaderName, Request, Response};
+use pin_project::pin_project;
+use std::future::Future;
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+use tower_layer::Layer;
+use tower_service::Service;
+
+/// Mark a header as [sensitive] on both requests and responses.
+///
+/// [sensitive]: https://docs.rs/http/latest/http/header/struct.HeaderValue.html#method.set_sensitive
+#[derive(Clone, Debug)]
+pub struct SetSensitiveHeaderLayer {
+    header: HeaderName,
+}
+
+impl SetSensitiveHeaderLayer {
+    /// Create a new [`SetSensitiveHeaderLayer`].
+    pub fn new(header: HeaderName) -> Self {
+        Self { header }
+    }
+}
+
+impl<S> Layer<S> for SetSensitiveHeaderLayer {
+    type Service = SetSensitiveRequestHeader<SetSensitiveResponseHeader<S>>;
+
+    fn layer(&self, inner: S) -> Self::Service {
+        SetSensitiveRequestHeader::new(
+            SetSensitiveResponseHeader::new(inner, self.header.clone()),
+            self.header.clone(),
+        )
+    }
+}
+
+/// Mark a request header as [sensitive].
```

Comment:
it might be nice if these docs included a link to the service this layer produces? not a big deal though.